Travis reports during `make check` (a.k.a src/test/test_bitcoin):
   
    2015-01-28 20:27:11 Unlocked: cs_Shutdown  init.cpp:141
    make[2]: *** [check-local] Error 1

It looks like test_bitcoin crashes at `init.cpp:141` (`Shutdown()`)
This like is `TRY_LOCK(cs_Shutdown, lockShutdown);` (https://github.com/bitcoin/bitcoin/blob/master/src/init.cpp#L141)

And this pull introduces a mutex.
I tend to NACK unless this is sorted out.

--- a/src/checkqueue.h
+++ b/src/checkqueue.h
@@ -161,7 +161,12 @@ public:
     {
     }
 
-    friend class CCheckQueueControl<T>;
+    bool IsIdle()
+    {
+        boost::unique_lock<boost::mutex> lock(mutex);
+        return (nTotal == nIdle && nTodo == 0 && fAllOk == true);
+    }
+
 };
 
 /** 
@@ -180,9 +185,8 @@ public:
     {
         // passed queue is supposed to be unused, or NULL
         if (pqueue != NULL) {
-            assert(pqueue->nTotal == pqueue->nIdle);
-            assert(pqueue->nTodo == 0);
-            assert(pqueue->fAllOk == true);
+            bool isIdle = pqueue->IsIdle();
+            assert(isIdle);
         }
     }
 

